### PR TITLE
487 option to delete feedbacks

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Added possibility to delete feedback in the feedbacks list view

## Related Issue

N/A

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: frontend/src/views/main/project/modals/ConfirmModal.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<template>
```
Comment (Googleton) [inline frontend/src/views/main/project/modals/ConfirmModal.vue:1]: For this file, we have a utility already, you can look in AdminUsers.vue line 137 for an example

File: frontend/src/views/main/project/FeedbackList.vue [hunk 2]
```diff
@@ -78,6 +78,19 @@
               (item.featureValue && item.featureValue.length > 140) ? item.featureValue.slice(0, 140) + "..." : item.featureValue
             }}</span>
         </template>
+        <template v-slot:item.action="{item}"
+                  v-slot:item.id="{item}">
+          <ConfirmModal
+              openMessage="Delete"
+              confirmMessage="Delete"
+              title="Delete feedback"
+              :text="`Are you sure that you want to delete the feedback for feature ${ item.featureName } permanantly?`"
```
Comment (Googleton) [inline frontend/src/views/main/project/FeedbackList.vue:87]:
```suggestion
              :text="`Are you sure that you want to delete the feedback for feature ${ item.featureName } permanently?`"
```

File: frontend/src/views/main/project/FeedbackList.vue [hunk 3]
```diff
@@ -86,123 +99,125 @@
...
+});
 
-  openFeedbackDetail = false
+const feedbacks = ref<FeedbackMinimalDTO[]>([]);
+const search = ref<String>('');
+const modelFilter = ref<String>('');
+const datasetFilter = ref<String>('');
+const typeFilter = ref<String>('');
+const groupByFeature = ref<Boolean>(false);
+const openFeedbackDetail = ref<Boolean>(false);
```
Comment (Googleton) [inline frontend/src/views/main/project/FeedbackList.vue:122]: string and boolean can be lowercase here to match how it's done elsewhere

File: frontend/src/views/main/project/modals/ConfirmModal.vue (hunk 1 above)
Comment (kevinmessiaen) [reply frontend/src/views/main/project/modals/ConfirmModal.vue:1]: I'm not sure how I can get the `$dialog` reference with the vue 3 syntaxe

Comment (andreybavt) [reply frontend/src/views/main/project/modals/ConfirmModal.vue:1]: Yeah it's true that `vuetify-dialog` doesn't support composition API. I started looking at `vue-final-modal` to replace it, but didn't finish the investigation. Kevin, could you maybe have a look too?
https://v2.vue-final-modal.org/dynamic-modal/?utm_source=pocket_reader

File: backend/src/test/java/ai/giskard/web/rest/WithMockGiskardUser.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+@Target({ ElementType.METHOD, ElementType.TYPE })
```
[the remark this answers is not in the document]
Comment (kevinmessiaen) [reply backend/src/test/java/ai/giskard/web/rest/WithMockGiskardUser.java:23]: This one set the Principal as an instance of `GiskardUser` while `@WithMockUser` does not provide such option. 

This allow is mostly required due to the method `SecurityUtils.extractUserId` that is required to check if user id is same as project owner or guest for certain tests.

**Comment by kevinmessiaen** [comment]
@andreybavt Took a look at vue-final-modal and it was pretty straightforward to use. We would require a bit of refactoring to migrate from previous modal to new but can be done progressively as the migration to the composition API.

**Comment by Googleton** [comment]
Regarding the $dialog option, I quickly discussed it with Andrey earlier and there's 2 other solutions we talked about:

- Vue 3 has provide/inject that would allow us to keep using it, but I couldn't find how to use it in Vue 2.7 so maybe explore if that's possible
- Provide $dialog at the window level so it's available everywhere

I was going to explore this during the week but if you want to have a look that's also fine by me. Let me know !

**Comment by kevinmessiaen** [comment]
> Regarding the $dialog option, I quickly discussed it with Andrey earlier and there's 2 other solutions we talked about:
> 
> * Vue 3 has provide/inject that would allow us to keep using it, but I couldn't find how to use it in Vue 2.7 so maybe explore if that's possible
> * Provide $dialog at the window level so it's available everywhere
> 
> I was going to explore this during the week but if you want to have a look that's also fine by me. Let me know !

If it's possible to inject the dialog reference it would be very int

**Comment by andreybavt** [comment]
@kevinmessiaen , did you have a chance to test Vue Final Modal's ["dynamic modals"](https://vue-final-modal.org/dynamic-modal/)? Does it allow to `await` until the modal is closed or we'd have to define callbacks?

**Comment by kevinmessiaen** [comment]
@andreybavt I used callbacks:

```
$vfm.show({
    component: ConfirmModal,
    bind: {
      title: 'Delete feedback',
      text: `Are you sure that you want to delete the feedback for feature '${feedback.featureName}' permanently?`,
      isWarning: true
    },
    on: {
      async confirm(close) {
        await api.deleteFeedback(feedback.id);
        await fetchFeedbacks();
        close();
      }
    }
  });
  ```
  
 Regarding await it looks like it have to be implemented in the future: https://github.com/vue-final/vue-final-modal/issues/193
  
  However we can take avantage of composition API and make our utility:
  ```
  export function useModal(component) {
     return async (bind) => {
        return new promise((resolve) => {
          $vfm.show({
            component,
            bind,
            on: {
              result: (result: DialogResult<RESULT>) => {
		resolve(result);
	      },
	      closed() {
		resolve({ action: "close" });
	      }
            }
          });
        }
     }
  }
  ```

**Comment by andreybavt** [comment]
Great, I just noticed that you've added vfm to this PR already 👍

**Pull request merged**